Automattic/woocommerce-payments
# Fix transaction search with translated terms

Fixes #2903 

#### Changes proposed in this Pull Request
When Order # or Subscription # term is translated, it breaks the transaction search as the separator is hardcoded to #.
Changes introduced here splits the search term using the translated text.
No new unit tests are introduced, existing tests still covers this change.

https://user-images.githubusercontent.com/6216000/134193290-2b8d671b-aa45-4d93-a63b-b7ee685f51c2.mov









#### Testing instructions




**Testing search by order #**
* On a site with existing WCPay transactions, install Loco Translate and change the translation of `Order #` to some other text (say `Order no.`) via Loco Translate -> Plugins -> WooCommerce Payments -> Editor -> [Search Order # and update its translation]. Note: Changing the language to English (Canada) as mentioned in #2903 did not work for me.
* Go to Payments > Transactions and search for an order number.
* Autocompleter should suggest an order with translated text for Order # from the first step.
* Ensure that clicking on the order number returns the relevant transaction

**Testing search by Subscription #**
* Same as above, translate `Subscription #` to some other text
* Search for a subscription number on Payments -> Transactions screen and wait for autocompleted results to appear.
* Ensure that selecting the autocompleted result leads to the correct transaction.
-------------------


**Post merge**
